### PR TITLE
Wire up all of the boolean options

### PR DESCRIPTION
**Pull request by ddevault**

### CODE REVIEW

File: osu.Game/Configuration/OsuConfigManager.cs [hunk 1]
```diff
@@ -177,6 +177,18 @@ protected override void InitialiseDefaults()
             Set(OsuConfig.Ticker, false);
             Set(OsuConfig.CompatibilityContext, false);
             Set(OsuConfig.CanForceOptimusCompatibility, true);
+            // TODO:
+            /*
+            if (sMyPcSucks)
```
Comment (peppy) [inline osu.Game/Configuration/OsuConfigManager.cs]: fwiw we probably won't need this.

Comment (SirCmpwn) [inline osu.Game/Configuration/OsuConfigManager.cs]: Take out the TODO?

Comment (peppy) [inline osu.Game/Configuration/OsuConfigManager.cs]: yeah, let's do that.

File: osu.Game/Overlays/Options/Gameplay/GeneralGameplayOptions.cs [hunk 2]
```diff
@@ -16,11 +20,24 @@ public GeneralGameplayOptions()
...
-                new BasicCheckBox { LabelText = "Show approach circle on first \"Hidden\" object" },
-                new BasicCheckBox { LabelText = "Scale osu!mania scroll speed with BPM" },
-                new BasicCheckBox { LabelText = "Remember osu!mania scroll speed per beatmap" },
+                keyOverlay = new CheckBoxOption { LabelText = "Always show key overlay" },
+                hiddenApproachCircle = new CheckBoxOption { LabelText = "Show approach circle on first \"Hidden\" object" },
+                scaleManiaScroll = new CheckBoxOption { LabelText = "Scale osu!mania scroll speed with BPM" },
+                rememberManiaScroll = new CheckBoxOption { LabelText = "Remember osu!mania scroll speed per beatmap" },
             };
+        }
+
```
Comment (peppy) [inline osu.Game/Overlays/Options/Gameplay/GeneralGameplayOptions.cs]: can you please fix whitespace trimming on save or something :x

**Pull request merged**